### PR TITLE
fix(communities)_: kick AC notification after control node device change

### PR DESCRIPTION
**Pull request by mprakhov**
Added `ResendAccountsClock` to `CommunityDescription`, which will indicate, that the member was soft-kicked and required to share his revealed accounts

Added soft kick. A soft kick is a kick, when a member is kicked, but will not receive a kick AC notification and the member will resend his revealed accounts

Soft kicks can happen on:
- on ownership change
- on the control node device change (if the new control node does not have member revealed addresses)

Fixed bug with not starting members reevaluation loop after promoting self as a control node

Impact: 
- promote self as a control node

Closes # https://github.com/status-im/status-desktop/issues/15341

### CODE REVIEW

**Review by jrainville — APPROVED** [review]
Looks good. Nice work

File: protocol/communities/community.go [hunk 1]
```diff
@@ -972,6 +972,38 @@ func (o *Community) RemoveUserFromOrg(pk *ecdsa.PublicKey) (*protobuf.CommunityD
 	return o.config.CommunityDescription, nil
 }
 
+func (o *Community) RemoveSpecificUsersFromOrg(membersToRemove map[string]*protobuf.CommunityMember) *CommunityChanges {
```
Comment (osmaczko) [inline protocol/communities/community.go]: Just to keep the naming consistent.
```suggestion
func (o *Community) RemoveUsersFromOrg(membersToRemove map[string]*protobuf.CommunityMember) *CommunityChanges {
```


Also, I would love to see `RemoveUserFromOrg` and `RemoveUsersFromOrg` being compatible, that is:
```

func (o *Community) RemoveUserFromOrg(pk *ecdsa.PublicKey) (*CommunityChanges, error) {
    return o.RemoveUsersFromOrg([]*ecdsa.PublicKey{pk})
}
```

Do you think it is doable / worth the effort?

File: protocol/communities/manager.go [hunk 2]
```diff
@@ -2281,6 +2284,8 @@ func (m *Manager) handleCommunityDescriptionMessageCommon(community *Community,
 		if changes.HasMemberLeft(pkString) {
 			// If we joined previously the community, that means we have been kicked
 			changes.MemberKicked = community.Joined()
+			// soft kick member on community owner change or on ResendAccountsClock change
+			changes.MemberSoftKicked = changes.ControlNodeChanged != nil || prevResendAccountsClock < community.Description().ResendAccountsClock
 		}
```
Comment (osmaczko) [inline protocol/communities/manager.go]: Member can be either kicked or soft kicked, but never both, right? I see the logic as follows:
```suggestion
		if changes.HasMemberLeft(pkString) && community.Joined() {
			if changes.ControlNodeChanged != nil || prevResendAccountsClock < community.Description().ResendAccountsClock {
				changes.MemberSoftKicked = true
			} else {
				changes.MemberKicked = true
			}
		}
```

File: protocol/communities_messenger_signers_test.go [hunk 3]
```diff
@@ -824,3 +844,156 @@ func (s *MessengerCommunitiesSignersSuite) TestWithoutMintedOwnerTokenMakingDevi
...
+	s.Require().Len(response.CommunityChanges, 1)
+	s.Require().Len(response.CommunityChanges[0].MembersRemoved, 0)
+
+	// Simulate control node device changed and new control node device does not have bob's revealed addresses
+	bobRequestID := communities.CalculateRequestID(s.bob.IdentityPublicKeyString(), community.ID())
+	err = s.john.communitiesManager.RemoveRequestToJoinRevealedAddresses(bobRequestID)
+	s.Require().NoError(err)
+
+	// due to test execution is fast, we update request to join clock
+	clock := uint64(time.Now().Unix() - 2)
```
Comment (osmaczko) [inline protocol/communities_messenger_signers_test.go:943]: Can we avoid that? I am pretty sure this test will be flaky.. Also,I am struggling to understand why it is needed?

File: protocol/messenger_communities.go [hunk 4]
```diff
@@ -4808,6 +4796,8 @@ func (m *Messenger) PromoteSelfToControlNode(communityID types.HexBytes) (*Messe
 		m.config.messengerSignalsHandler.MessengerResponse(&response)
 	}
 
+	go m.communitiesManager.StartMembersReevaluationLoop(community.ID(), false)
```
Comment (osmaczko) [inline protocol/messenger_communities.go]:
```suggestion
	m.communitiesManager.StartMembersReevaluationLoop(community.ID(), false)
```

**Review by osmaczko — COMMENTED** [review]
Thanks for the PR. I have few suggestions/concerns.

File: protocol/communities/manager.go (hunk 2 above)
Comment (mprakhov) [reply protocol/communities/manager.go]: I think it depends on how you threat these flags.
You can say, that the member is kicked or softKicked (MemberKicked || MemberSoftKicked) or you can say that the member is kicked and it is soft kick (MemberKicked && MemberSoftKicked)

It's not a problem to make MemberKicked || MemberSoftKicked only and both can't exist

File: protocol/communities_messenger_signers_test.go (hunk 3 above)
Comment (mprakhov) [reply protocol/communities_messenger_signers_test.go:943]: No, and it is not flaky, this method is used in other test scenarios
The problem is that if the user does not have revealed addresses, we kick him and generate a mock request to join with AwaitingForSharedAddress status for automatic rejoin (`GenerateRequestsToJoinForAutoApprovalOnNewOwnership`). This updates the clock.
All this happens during `PromoteSelfToControlNode`.
Logic of saving updating of RequestsToJoin is that clock must be different, but between first and second execution of `PromoteSelfToControlNode` took less than 1 sec

File: protocol/communities/community.go (hunk 1 above)
Comment (mprakhov) [reply protocol/communities/community.go]: I don't have a strong opinion here. From the code reading complexity, removeMemberFromOrg is much easier to read :) 
Not a problem to refactor

Comment (mprakhov) [reply protocol/communities/community.go]: refactored, changes in the separate commit

File: protocol/communities/manager.go (hunk 2 above)
Comment (mprakhov) [reply protocol/communities/manager.go]: refactored, but in a little bit another way :)

**Review by osmaczko — APPROVED** [review]
Thanks for addressing comments!

**Pull request merged**